src/address.ts

```diff
@@ -159,15 +159,32 @@ export class Address {
       '{{name.lastName}}{{address.citySuffix}}',
     ];
 
-    if (!format && this.faker.definitions.address.city_name) {
+    if (this.faker.definitions.address.city_name != null) {
       formats.push('{{address.cityName}}');
     }
 
-    if (typeof format !== 'number') {
-      format = this.faker.datatype.number(formats.length - 1);
-    }
+    const useFormat = (() => {
+      if (typeof format === 'number') {
+        deprecated({
+          deprecated: 'address.city(number)',
```

Comment:
On all places where we use this function, we use `deprecated: 'faker.address.city(number)'`

```suggestion
          deprecated: 'faker.address.city(number)',
```

Also, we may provide a hint what the user should use instead.

Reply:
FYI:
https://github.com/faker-js/faker/blob/712b1de126ea6580660a320e065c35ac775f09b3/src/name.ts#L30-L35
https://github.com/faker-js/faker/blob/712b1de126ea6580660a320e065c35ac775f09b3/src/helpers.ts#L510-L515
https://github.com/faker-js/faker/blob/712b1de126ea6580660a320e065c35ac775f09b3/src/helpers.ts#L584-L588
https://github.com/faker-js/faker/blob/712b1de126ea6580660a320e065c35ac775f09b3/src/helpers.ts#L636-L641

Reply:
done in 0747cb4aa2a2d0b6e49650019a018767ab8b29eb